src/suggestion.rs

```diff
@@ -50,14 +69,206 @@ impl fmt::Display for Detector {
+            .take(mistake_range.len())
+            .collect();
+        (marker_size, full)
+    };
+
+    // calculate the available space after accounting for the static and shrunken mistake
+    let avail_space = terminal_size
+        .saturating_sub(marker_size + padding_till_excerpt_start + TOTAL_CONTEXT_CHAR_COUNT);
+
+    // take both sides of the mistake and insert the possibly shrunken mistake
+    // and put them together, after conditioning the left and right context
+    let stripped_line_len = stripped_line.chars().count();
+
+    // full, uncut context coverage
+    let left_context = Range {
+        start: 0,
+        end: mistake_range.start,
+    };
+    let right_context = Range {
+        start: mistake_range.end,
+        end: stripped_line_len,
+    };
+
+    let stripped_line_len = stripped_line.chars().count();
+    // left and right we account half of it
+    let avail_space_half = avail_space / 2usize;
+
+    let (left_context, right_context) = match (
+        avail_space_half > left_context.len(),
+        avail_space_half > right_context.len(),
+    ) {
+        (true, false) => {
+            // left context does not use all the capacity avail
+            // allow the right context to consume the excess.
+            let right_avail_space = avail_space - left_context.len();
+            (
+                left_context,
+                Range {
+                    start: right_context.end,
+                    end: cmp::min(mistake_range.end + right_avail_space, stripped_line_len),
+                },
+            )
+        }
+        (false, true) => {
+            // right context does not use all the capacity avail
+            // allow the left context to consume the excess.
+            let left_avail_space = avail_space - right_context.len();
+            (
+                Range {
+                    start: left_context.end.saturating_sub(left_avail_space),
+                    end: left_context.end,
+                },
+                right_context,
+            )
+        }
+        (false, false) => {
+            // both sides have excess chars, so yield `avail_space_half` to both sides
+            (
+                Range {
+                    start: left_context.end.saturating_sub(avail_space_half),
+                    end: left_context.end,
+                },
+                Range {
+                    start: right_context.start,
+                    end: right_context.start + avail_space_half,
+                },
+            )
+        }
+        _ => {
+            // both sides are less than the allowed context, no need to modify
+            (left_context, right_context)
+        }
+    };
+
+    assert!(left_context.end == mistake_range.start);
+    assert!(right_context.end <= stripped_line_len);
+    assert!(left_context.len() + mistake_range.len() + right_context.len() <= stripped_line_len);
+
+    let more_left = if left_context.start == 0 { "" } else { ".." };
```

Comment:
this is great!

Reply:
Just felt visually "better" and yields two characters we can use for extra content - good point that you accounted in the 10 for  those two! :+1: